src/scripts/web3-interface.ts

```diff
@@ -1,90 +1,58 @@
 import { exit } from "process";
 import { archLogger } from "../logger/chalk-theme";
-import {
-  IERC20,
-  ArchaeologistFacet__factory,
-  ViewStateFacet__factory,
-  ViewStateFacet,
-  IERC20__factory,
-  ThirdPartyFacet,
-  ThirdPartyFacet__factory,
-  EmbalmerFacet,
-  EmbalmerFacet__factory,
-} from "@sarcophagus-org/sarcophagus-v2-contracts";
 import { BAD_ENV } from "../utils/exit-codes";
-import { getNetworkConfigByChainId, localChainId } from "../lib/config";
-import { NetworkConfig } from "../lib/types/network-config";
+import { NetworkContext, getNetworkContextsByChainIds } from "../network-config";
 import { KeyFinder } from "../models/key-finder";
-import { ArchaeologistFacetX } from "./web3-interface/archaeologist-facet-x";
+import { SarcoSupportedNetwork } from "@sarcophagus-org/sarcophagus-v2-sdk";
 
 export interface Web3Interface {
-  networkName: string;
-  ethWallet: ethers.Wallet;
-  encryptionHdWallet: ethers.utils.HDNode;
-  keyFinder: KeyFinder;
-  sarcoToken: IERC20;
-  archaeologistFacet: ArchaeologistFacetX;
-  embalmerFacet: EmbalmerFacet;
-  thirdPartyFacet: ThirdPartyFacet;
-  viewStateFacet: ViewStateFacet;
-  networkConfig: NetworkConfig;
+  networkContexts: NetworkContext[];
+  getNetworkContext: (network: SarcoSupportedNetwork | undefined) => NetworkContext;
 }
 
 let web3Interface: Web3Interface | undefined;
 
 export const destroyWeb3Interface = async (): Promise<void> => {
   if (!!web3Interface) {
-    web3Interface.ethWallet.provider.removeAllListeners();
-    (web3Interface.ethWallet.provider as ethers.providers.WebSocketProvider)._websocket.terminate();
+    web3Interface.networkContexts.forEach(networkContext => {
+      networkContext.ethWallet.provider.removeAllListeners();
+      (
+        networkContext.ethWallet.provider as ethers.providers.WebSocketProvider
+      )._websocket.terminate();
+    });
     web3Interface = undefined;
   }
 };
 
-export const getWeb3Interface = async (isTest?: boolean): Promise<Web3Interface> => {
+export const getWeb3Interface = async (isTest: boolean = false): Promise<Web3Interface> => {
   if (!!web3Interface) {
     return web3Interface;
   }
 
   try {
-    const networkConfig = getNetworkConfigByChainId(process.env.CHAIN_ID || localChainId);
-    // PROVIDER_URL should be "wss://<network>.infura.io/ws/v3/<api-key>"
-    const rpcProvider = new ethers.providers.WebSocketProvider(process.env.PROVIDER_URL!);
-
-    const ethWallet = isTest
-      ? ethers.Wallet.createRandom()
-      : new ethers.Wallet(process.env.ETH_PRIVATE_KEY!, rpcProvider);
-
-    const encryptionHdWallet = ethers.utils.HDNode.fromMnemonic(process.env.ENCRYPTION_MNEMONIC!);
-
-    const network = await rpcProvider.detectNetwork();
-
-    const sarcoToken = IERC20__factory.connect(networkConfig.sarcoTokenAddress, ethWallet);
+    const chainIds = process.env.CHAIN_IDS
+      ? process.env.CHAIN_IDS.split(",").map(chaidIdStr => Number(chaidIdStr.trim()))
+      : [];
```

Comment:
Should we allow an empty array here? Seems like the arch program should exit if there are no chain IDs